fiftyone/brain/internal/core/leaky_splits.py

```diff
@@ -0,0 +1,497 @@
+"""
+Finds leaks between splits.
+"""
+
+from collections import defaultdict
+from copy import copy
+
+import fiftyone as fo
+from fiftyone import ViewField as F
+
+# pylint: disable=no-member
+import cv2
+
+import eta.core.utils as etau
+
+import fiftyone.core.brain as fob
+import fiftyone.brain.similarity as sim
+import fiftyone.brain.internal.core.sklearn as skl_sim
+import fiftyone.brain.internal.core.duplicates as dups
+import fiftyone.brain.internal.core.utils as fbu
+import fiftyone.core.utils as fou
+import fiftyone.core.validation as fov
+import fiftyone.zoo as foz
+
+fbu = fou.lazy_import("fiftyone.brain.internal.core.utils")
+
+_DEFAULT_MODEL = "clip-vit-base32-torch"
+_DEFAULT_BATCH_SIZE = None
+
+
+def compute_leaky_splits(
+    samples,
+    brain_key,
+    split_views=None,
+    split_field=None,
+    split_tags=None,
+    threshold=0.2,
+    embeddings_field=None,
+    model=None,
+    model_kwargs=None,
+    patches_field=None,
+    metric="cosine",
+    **kwargs,
```

Comment:
There's no way to chose the hash-based backend right? That seems missing

Reply:
Also, this method would not ever interface with/act on patches, right? Can safely remove that from explicitly in function signatures

Reply:
@brimoor Hash based backend is in a separate class, `LeakySplitsHash`. @jacobmarks I'll remove the patches stuff.

Reply:
I see that it is a separate class, but how do you actually tell FOB you want to use that method when you call `compute_leaky_splits()`?

Reply:
`compute_leaky_splits` doesn't support the hash method. I can extend it to do so. The hash based method seems to not be very useful because it looks for exact duplicates, something already covered by find exact duplicates and something that most people already have checks in place for. If it's important for it to be in the main function I'll add it

Reply:
Either it is useful and should be available via `compute_leaky_splits()` or it is not useful and should just be deleted from the codebase IMO. Adding unused code is just more work/confusion to maintain in the future.

Reply:
I lean towards removing it. Please let me know what you think.

Reply:
Suggested path forward:
1. remove hashing from leaky-splits
2. Backlog: add new hash function (image hash) as an option for `compute_duplicates`